game-core/src/main/java/games/strategy/triplea/ui/DicePanel.java

```diff
@@ -44,15 +46,22 @@ public void setDiceRoll(final DiceRoll diceRoll) {
     SwingAction.invokeNowOrLater(() -> {
       removeAll();
       for (int i = 1; i <= data.getDiceSides(); i++) {
-        final List<Die> dice = diceRoll.getRolls(i);
+        final int index = i;
+        final List<Die> dice = Optional.ofNullable(diceRoll)
+            .map(roll -> roll.getRolls(index))
+            .orElse(Collections.emptyList());
         if (dice.isEmpty()) {
           continue;
         }
         add(new JLabel("Rolled at " + (i) + ":"));
         add(create(diceRoll.getRolls(i)));
       }
       add(Box.createVerticalGlue());
-      add(new JLabel("Total hits:" + diceRoll.getHits()));
+      final String hitCount = Optional.ofNullable(diceRoll)
+          .map(DiceRoll::getHits)
+          .map(String::valueOf)
+          .orElse("0");
```

Comment:
This can be shortened, theres no need to explicitly convert a number to a string, using an int will do.

Reply:
Done: ff88eff